internal/transport/transport_test.go

```diff
@@ -2752,9 +2840,8 @@ func (s) TestClientDecodeHeaderStatusErr(t *testing.T) {
 			}
 
 			s.operateHeaders(test.metaHeaderFrame)
-
 			got := ts.status
-			want := test.wantStatus
+			want := test.wantStatusEndStream
```

Comment:
This test is feeling pretty complicated to me now.  Would it be better to just split this test into two separate test functions since it's running two different tests and has multiple different assertions?